### ISSUE TITLE
Move segment configuration out of base template

### ISSUE THREAD

**Issue by mechanomi** [crate/crate-docs-theme#190]
there are a few third-party sites that are using an adapted version of this theme. unfortunately, because we embed our segment tracking directly into the HTML, this is segment tracking code is being executed on these third-party sites, which is throwing off our analytics

@autophagy suggests the following:

Would it not be possible to just remove it from here: https://github.com/crate/crate-docs-theme/blob/54ae114b7f63d7e7cdd43d827418dcd8fd50ea54/src/crate/theme/rtd/conf/__init__.py#L57

and instead, have it an option in the conf.py files that each project has? Like: https://github.com/autophagy/faereld/blob/master/seonu/conf.py

For example, I just changed the conf.py file for the crate-howtos repo (https://github.com/crate/crate-howtos/blob/master/docs/conf.py) to:

```
from crate.theme.rtd.conf.crate_howtos import *

html_theme_options = {
    'tracking_segment_id': 'hello world!'
}
```

and built the project, which resulted in this in the source:

![image](https://user-images.githubusercontent.com/23469/84873314-56b30a80-b083-11ea-9148-daeb791fcfce.png)

**Comment by mechanomi** [comment]
cc @msbt

**Comment by msbt** [comment]
@mechanomi thanks for pushing this forward. since this is a very crate specific setting, it would be nice to have the segment snippets (one time initialization, one tracking event when the feedback box is being loaded) only built in if it's actually a project of our own, else people are loading the analytics.js for no reason. could we - in addition to adding the segment id via conf.py - wrap the two snippets around something like

```
{% if crate_docs %}
segment stuff
{% endif %}
```
so third parties that aren't using segment don't have to load the 350kb of external javascript?

**Comment by jodok** [comment]
let's just put the actual code out of github and seed it in the build process!

**Comment by autophagy** [comment]
I've made a PR that does the following thing, based on @jodok's suggestion:
- Removes the tracking code from the theme
- Allows us to set the tracking code during the build phase, by default, using an environment variable ``TRACKING_SEGMENT_ID``. This lets us set it in the RTD projects themselves, rather than in the code of the project's configuration.

Once this is merged & we've released the docs theme, we should invalidate the old tracking id code and add the new one to the RTD projects with the environment variable.

**Comment by autophagy** [comment]
The crate-docs-theme is now released with version 0.10.0 that contains this feature. I've added the new tracking code as an environment variable to all the RTD projects and issued a mass rebuild. The new key is now showing up on our documentation, so Matthias has revoked the old one.
